clients/feeder/feeder.go

```diff
@@ -249,11 +256,15 @@ func (c *Client) get(ctx context.Context, queryURL string) (io.ReadCloser, error
 				req.Header.Set("X-Throttling-Bypass", c.apiKey)
 			}
 
+			c.client.Timeout = c.timeouts[c.curTimeout]
 			reqTimer := time.Now()
 			res, err = c.client.Do(req)
 			if err == nil {
 				c.listener.OnResponse(req.URL.Path, res.StatusCode, time.Since(reqTimer))
 				if res.StatusCode == http.StatusOK {
+					if c.curTimeout > 0 {
+						c.curTimeout--
+					}
```

Comment:
This new lines should be all logic managed by the `TimeoutList` type

Reply:
moved